### PR TITLE
Fix actors added by rule on Ticket update

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #10564

### CODE REVIEW

**Comment by trasher** [comment]
I guess this would need a test

**Review by cedric-anne — COMMENTED** [review]
It seems related to #10739.

**Review by cedric-anne — CHANGES_REQUESTED** [review]
Unless I am wrong, it will not handle groups and suppliers.

File: src/CommonITILObject.php [hunk 1]
```diff
@@ -8139,7 +8139,25 @@ protected function updateActors()
...
+
+                if ($existing['itemtype'] === 'User') {
+                    $input_field_name = '_additional_' . $actortype_str . 's';
+                } else {
+                    $type = strtolower($existing['itemtype']::getType());
+                    $input_field_name = '_additional_' . $type . 's_' . $actortype_str . 's';
+                }
+
+                if (isset($this->input[$input_field_name]) && count($this->input[$input_field_name])) {
+                    $first_item = reset($this->input[$input_field_name]);
```
Comment (cedric-anne) [inline src/CommonITILObject.php:8151]: Could you please add a comment in code to explain what are expected formats ? I had to read twice to be able to understand.

File: src/CommonITILObject.php [hunk 2]
```diff
@@ -8139,7 +8139,25 @@ protected function updateActors()
...
+                    if (!is_array($first_item)) {
+                        $input_ids = isset($this->input[$input_field_name]) ? array_values($this->input[$input_field_name]) : [];
```
Comment (cedric-anne) [inline src/CommonITILObject.php:8153]: `isset` check is done on line 8150.
```suggestion
                        $input_ids = array_values($this->input[$input_field_name]);
```

File: src/CommonITILObject.php [hunk 3]
```diff
@@ -8139,7 +8139,25 @@ protected function updateActors()
...
-                if ($found === false) {
+
+                if ($existing['itemtype'] === 'User') {
+                    $input_field_name = '_additional_' . $actortype_str . 's';
+                } else {
+                    $type = strtolower($existing['itemtype']::getType());
+                    $input_field_name = '_additional_' . $type . 's_' . $actortype_str . 's';
+                }
+
+                if (isset($this->input[$input_field_name]) && count($this->input[$input_field_name])) {
```
Comment (cedric-anne) [inline src/CommonITILObject.php:8150]:
```suggestion
                if (isset($this->input[$input_field_name]) && is_array($this->input[$input_field_name]) && count($this->input[$input_field_name])) {
```

File: src/CommonITILObject.php [hunk 4]
```diff
@@ -8139,7 +8139,25 @@ protected function updateActors()
...
+                    $first_item = reset($this->input[$input_field_name]);
+                    if (!is_array($first_item)) {
+                        $input_ids = isset($this->input[$input_field_name]) ? array_values($this->input[$input_field_name]) : [];
+                    } else {
+                        $input_ids = isset($this->input[$input_field_name]) ? array_column($this->input[$input_field_name], $existing['itemtype']::getForeignKeyField()) : [];
```
Comment (cedric-anne) [inline src/CommonITILObject.php:8155]:
```suggestion
                        $input_ids = array_column($this->input[$input_field_name], $existing['itemtype']::getForeignKeyField());
```

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**